sources/core/objects/turfs/MetalWall.h

```diff
@@ -1,26 +1,47 @@
 #pragma once
 
 #include "Turf.h"
+#include <bitset>
+#include <array>
 
 namespace kv
 {
 
+const std::array<Dir, 4> WALL_PROCESSING_DIRS = {Dir::NORTH, Dir::SOUTH, Dir::WEST, Dir::EAST};
+
 class MetalWall : public Turf
 {
 public:
     DECLARE_SAVEABLE(MetalWall, Turf);
     REGISTER_CLASS_AS(MetalWall);
+
     MetalWall();
+
+    void UpdateState(Dir dir = Dir::ALL);
+    void UpdateNeighborhoodState(Dir dir = Dir::ALL);
+    void CheckNeighborhood(Dir dir = Dir::ALL);
+
     virtual void AttackBy(IdPtr<Item> item) override;
+
+    virtual void AfterWorldCreation() override;
+    virtual void Delete() override;
+protected:
+    QString default_state_;
+    std::bitset<4> current_state_;
```

Comment:
All `kv::Object` successors should use `KV_SAVEABLE` macro on their member variables.
With help of that macro a certain reflection and serialization features are implementated, so without it multi-player game won't work correctly (map saves/loads for example).
`std::bitset` is kinda correct choice here, but unfortunately it will be pretty bothersome to write serialization functions for that class, so it would be much easier to just use `qint32` here or something.

Reply:
`std::bitset` is more elegant than `int` because it have `set` method.